tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -3263,6 +3263,7 @@ void EnqueueProgramImpl(
     detail::CompileProgram(device, program);
     program.allocate_circular_buffers(device);
     detail::ValidateCircularBufferRegion(program, device);
+    program.call_pre_exec_callback();
```

Comment:
I needed to be able to execute code after the CB allocation was done, but before the program object was deleted.

Reply:
rather than logging the max cb usage in this manner could it be added as part of the memory reports that are generated. it could be an additional CB memory report. 

alternatively could this be something that graph capture could support @ayerofieiev-tt

Reply:
Program exposes access to circular buffers on a particular core range. could inspecting a particular programs CBs before execution be done at op level?